quinn-proto/src/connection/mod.rs

```diff
@@ -1653,6 +1653,37 @@ where
         Ok(())
     }
 
+    /// Set the priority of a stream
+    ///
+    /// # Panics
+    /// - when applied to a receive stream
```

Comment:
Would it be reasonable to return `UnknownStream` in this case (and below)? We generally prefer not to panic even under API misuse, since the caller might be able to isolate and/or report failures more gracefully even when they're caused by caller bugs. Maybe tweak the comment to "... priority of an outgoing stream" for clarity.

Reply:
Would that not be inconsistent with `reset`? That function also panics when applied to a receive stream and that's also where I copied the assertion from.

Reply:
Good question. I suppose we haven't thought too hard about this in the past since the `quinn` layer enforces sanity at the type level. Given the precedent I'm happy to revisit this if/when we change reset's behavior.